File: publication-figure-renderer/r/tests/testthat/test-config.R
VALID_UUID <- "11111111-1111-1111-1111-111111111111"

test_that("accepts a minimal argv with --output and defaults sensibly", {
  config <- load_config(
    c(VALID_UUID, "--output", "fig.svg"),
    list(EVAGENE_API_KEY = "evg_test")
  )

  expect_equal(config$base_url, "https://evagene.net")
  expect_equal(config$api_key, "evg_test")
  expect_equal(config$pedigree_id, VALID_UUID)
  expect_equal(config$output_path, "fig.svg")
  expect_false(config$deidentify)
  expect_equal(config$label_style, "generation-number")
  expect_null(config$width)
  expect_null(config$height)
})

test_that("honours a custom base URL", {
  config <- load_config(
    c(VALID_UUID, "--output", "fig.svg"),
    list(EVAGENE_API_KEY = "evg_test", EVAGENE_BASE_URL = "https://evagene.example")
  )
  expect_equal(config$base_url, "https://evagene.example")
})

test_that("parses --deidentify and --label-style", {
  config <- load_config(
    c(VALID_UUID, "--output", "fig.svg", "--deidentify", "--label-style", "initials"),
    list(EVAGENE_API_KEY = "evg_test")
  )
  expect_true(config$deidentify)
  expect_equal(config$label_style, "initials")
})

test_that("parses --width and --height as positive integers", {
  config <- load_config(
    c(VALID_UUID, "--output", "fig.svg", "--width", "800", "--height", "600"),
    list(EVAGENE_API_KEY = "evg_test")
  )
  expect_equal(config$width, 800L)
  expect_equal(config$height, 600L)
})

test_that("missing API key raises config_error", {
  expect_error(
    load_config(c(VALID_UUID, "--output", "fig.svg"), list()),
    class = "config_error"
  )
})

test_that("missing --output raises config_error", {
  expect_error(
    load_config(c(VALID_UUID), list(EVAGENE_API_KEY = "evg_test")),
    class = "config_error"
  )
})

test_that("pedigree id must be a UUID", {
  expect_error(
    load_config(
      c("not-a-uuid", "--output", "fig.svg"),
      list(EVAGENE_API_KEY = "evg_test")
    ),
    class = "config_error"
  )
})

test_that("unknown --label-style raises config_error", {
  expect_error(
    load_config(
      c(VALID_UUID, "--output", "fig.svg", "--label-style", "anagram"),
      list(EVAGENE_API_KEY = "evg_test")
    ),
    class = "config_error"
  )
})

test_that("non-integer --width raises config_error", {
  expect_error(
    load_config(
      c(VALID_UUID, "--output", "fig.svg", "--width", "tall"),
      list(EVAGENE_API_KEY = "evg_test")
    ),
    class = "config_error"
  )
})

File: publication-figure-renderer/python/tests/test_config.py
import pytest

from publication_figure_renderer.config import (
    ConfigError,
    LabelStyle,
    load_config,
)

_VALID_UUID = "11111111-1111-1111-1111-111111111111"


def _minimal_env() -> dict[str, str]:
    return {"EVAGENE_API_KEY": "evg_test"}


def test_defaults_to_generation_number_style_without_deidentify() -> None:
    config = load_config([_VALID_UUID, "--output", "fig.svg"], _minimal_env())

    assert config.pedigree_id == _VALID_UUID
    assert config.output_path == "fig.svg"
    assert config.deidentify is False
    assert config.label_style is LabelStyle.GENERATION_NUMBER
    assert config.width is None
    assert config.height is None
    assert config.base_url == "https://evagene.net"


def test_parses_deidentify_and_label_style() -> None:
    config = load_config(
        [_VALID_UUID, "--output", "fig.svg", "--deidentify", "--label-style", "initials"],
        _minimal_env(),
    )
    assert config.deidentify is True
    assert config.label_style is LabelStyle.INITIALS


def test_parses_width_and_height() -> None:
    config = load_config(
        [_VALID_UUID, "--output", "fig.svg", "--width", "800", "--height", "600"],
        _minimal_env(),
    )
    assert config.width == 800
    assert config.height == 600


def test_honours_custom_base_url() -> None:
    config = load_config(
        [_VALID_UUID, "--output", "fig.svg"],
        {"EVAGENE_API_KEY": "evg_test", "EVAGENE_BASE_URL": "https://evagene.example"},
    )
    assert config.base_url == "https://evagene.example"


def test_missing_api_key_raises() -> None:
    with pytest.raises(ConfigError):
        load_config([_VALID_UUID, "--output", "fig.svg"], {})


def test_missing_output_raises() -> None:
    with pytest.raises(ConfigError):
        load_config([_VALID_UUID], _minimal_env())


def test_non_uuid_pedigree_id_raises() -> None:
    with pytest.raises(ConfigError):
        load_config(["not-a-uuid", "--output", "fig.svg"], _minimal_env())


def test_unknown_label_style_raises() -> None:
    with pytest.raises(ConfigError):
        load_config(
            [_VALID_UUID, "--output", "fig.svg", "--label-style", "anagram"],
            _minimal_env(),
        )


def test_non_positive_width_raises() -> None:
    with pytest.raises(ConfigError):
        load_config(
            [_VALID_UUID, "--output", "fig.svg", "--width", "0"],
            _minimal_env(),
        )
